lib/parser/provenance.cc

```diff
@@ -328,6 +360,43 @@ std::optional<ProvenanceRange> CookedSource::GetProvenanceRange(
   return {ProvenanceRange{first.start(), last.start() - first.start()}};
 }
 
+std::optional<CharBlock> CookedSource::GetCharBlockFromLineAndColumns(
+    int line, int column1, int column2) const {
+  auto provenanceStart = allSources_.GetFirstFileProvenance().value().start();
+  auto sourceFile = allSources_.GetSourceFile(provenanceStart);
+  if (!sourceFile) {
+    return std::nullopt;
+  }
+  std::optional<std::size_t> firstOffset =
```

Comment:
A nest of `if (auto firstOffset{...}) { if (auto secondOffset{...}) { ...` would be more clear.  You're setting up a sequence of hurdles that all have to be true in order to get to a non-empty `return`.

Reply:
Is this what you had in mind? I figured that too much if nesting would clog up the code, but I understand that this is easier to understand.